playback/src/audio_backend/alsa.rs

```diff
@@ -205,20 +209,24 @@ impl Sink for AlsaSink {
     }
 
     fn stop(&mut self) -> io::Result<()> {
-        {
-            // Write any leftover data in the period buffer
-            // before draining the actual buffer
-            self.write_bytes(&[])?;
-            let pcm = self.pcm.as_mut().ok_or_else(|| {
-                io::Error::new(io::ErrorKind::Other, "Error stopping AlsaSink, PCM is None")
-            })?;
-            pcm.drain().map_err(|e| {
+        // Write any leftover data in the period buffer
+        // before draining the actual buffer
+        self.drain_period_buffer()?;
+        match self.pcm.as_mut() {
+            Some(pcm) => pcm.drain().map_err(|e| {
                 io::Error::new(
                     io::ErrorKind::Other,
                     format!("Error stopping AlsaSink {}", e),
                 )
-            })?
-        }
+            })?,
+            None => {
+                return Err(io::Error::new(
+                    io::ErrorKind::NotFound,
+                    "Error stopping AlsaSink, PCM is None",
+                ))
+            }
+        };
```

Comment:
I prefer `if let Some(x)` syntax when there it's either `Some` or `None` and no more variants.

Reply:
I can do that.

Reply:
I just reverted it to the way it was.